frontend/src/lib/logic/apiStatusLogic.ts

```diff
@@ -0,0 +1,52 @@
+import { actions, kea, listeners, path, reducers } from 'kea'
+import api from 'lib/api'
+
+import type { apiStatusLogicType } from './apiStatusLogicType'
+
+export const apiStatusLogic = kea<apiStatusLogicType>([
+    path(['lib', 'apiStatusLogic']),
+    actions({
+        onApiResponse: (response?: Response, error?: unknown) => ({ response, error }),
+        setInternetConnectionIssue: (issue: boolean) => ({ issue }),
+    }),
+    reducers({
+        internetConnectionIssue: [
+            false,
+            {
+                setInternetConnectionIssue: (_, { issue }) => issue,
+            },
+        ],
+    }),
+    listeners(({ cache, actions, values }) => ({
+        onApiResponse: async ({ response, error }, breakpoint) => {
+            if (error || !response?.status) {
+                // Likely CORS headers errors (i.e. request failing without reaching Django))
+                if ((error as any)?.message === 'Failed to fetch') {
+                    actions.setInternetConnectionIssue(true)
```

Comment:
i wonder if we should breakpoint(50) here to avoid flapping this status?

Reply:
Good thinking!